ui/app/AppLayouts/Communities/views/EditCommunityTokenView.qml

```diff
@@ -264,14 +270,13 @@ StatusScrollView {
             extraValidator.errorMessage: qsTr("Enter a number between 1 and 999,999,999")
 
             onTextChanged: {
-                const amount = parseInt(text)
-                if (Number.isNaN(amount) || Object.values(errors).length)
+                const supplyNumber = parseInt(text)
+                if (Number.isNaN(supplyNumber) || Object.values(errors).length)
                     return
 
-                if(root.isAssetView)
-                    asset.supply = amount
-                else
-                    collectible.supply = amount
+                const token = root.isAssetView ? root.asset : root.collectible
+                token.supply = SQUtils.AmountsArithmetic.fromNumber(
```

Comment:
This looks wrong; the old code would assign to toplevel props (`asset` or `collectible`) but the new one just modifies a local variable?

Reply:
no, `token` is a reference, it works exactly as before